types/wlr_wl_shell.c

```diff
@@ -108,10 +109,10 @@ static void wl_shell_destroy(struct wl_resource *resource) {
 	}
```

Comment:
This is one of the "browse list to remove" pattern I don't see the point of.
resource is being remove, so even if it's not part of this list it should be removed from any list it belongs to.
Not sure how to check if someone else just removed the element from the list without adding it anywhere else though, but I don't know if we should expect users to mess with this list more than just read-looping over it like we do in handle_output_frame